### DISCUSSION TITLE
Exporting songs and skins from osu!lazer to a osu!stable format

### DISCUSSION THREAD

**Discussion by Skubers** [ppy/osu#23457] [Ideas] [answered]
I'm a mapper, and my hard drive recently died, and with it all of my beatmaps (these non-submitted too), skins, etc.
I noticed that some time ago I've imported all of the stable stuff to osu!lazer, but I can't seem to find any way to export it to regular files or import it to osu!stable. I prefer working on my maps on osu!stable, thus I wanted to export these files there.
Is there a way that I'm not seeing or should it be a feature?

**Comment by Theighlin** [comment]
You can only do it 1 by 1, and if you used certain curve types in the lazer editor the beatmap will not be exported properly i think.
Skins: Go to the skins section in the settings and click "export selected skin".
Beatmaps; Open them in the editor, File -> Export Package

**Comment by JesFEREM** [comment]
this is what I've been wanting for a while. also some games like McOsu may only support importing from stable, so having an option to have a separate stable install for stable-based third party games that happen to use it would be nice

**Comment by peppy** [comment, marked as answer]
For any kinda of batch exporting to a stable format, you'll need to rely on third party tools. This isn't something we will ever support first-party (but there is at least one tool in existence which can already do this for you).

**Reply by JesFEREM** [reply]
Apparently I was going insane and just didn't see the  map exporter that was the first thing on google
Last time I was looking for it I only found one that was last updated like 3 years ago that didn't work

**Reply by PetHedgehog** [reply]
hey, by any chance are you using the kabiiQ's beatmap exporter? it seems it cant be used for now until it gets updated, do you know how one would revert back an update on lazer? i tried removing the app-2023.12.18.0 folder and it works but no beatmaps get recognized then and i cant export them

**Reply by Theighlin** [reply]
Did you read the notification it showed you when you downgraded?
![image](https://github.com/ppy/osu/assets/53081750/6b6210e3-f433-4dff-ba7b-16430b46947c)
Go to the osu folder, rename client_newer_version.realm to client.realm and restart the game with the 2023.1218.0 update.

**Reply by PetHedgehog** [reply]
hmm i didnt see that second notif, ill try that

**Reply by PetHedgehog** [reply]
this doesnt seem to work, still empty library when trying to launch on 1130.0

**Reply by PetHedgehog** [reply]
![image](https://github.com/ppy/osu/assets/150826274/0c451550-5060-4562-bf4e-baf1a8b35413)
well deleting realm got me to have the message disappear but the library is still empty

**Reply by Theighlin** [reply]
Did you read the part where i said to restart the game with the new version? I was helping you get you maps back, not downgrading the game (you can do it by fixing the realm database, but i don't know what changed so i can't help you there).

**Reply by PetHedgehog** [reply]
oh well i know how to get it back, i just put back the app folder

**Reply by PetHedgehog** [reply]
ok now i think i actually messed it up, tried to install 1130.0 again and i thought it would do a separate install, now my library is actually gone
